### PR TITLE
Documentation and minor improvements for the Pool object

### PR DESCRIPTION
**Pull request by ajsantander**

### CODE REVIEW

File: packages/synthetix-main/contracts/modules/core/PoolModule.sol [hunk 1]
```diff
@@ -26,10 +25,6 @@ contract PoolModule is IPoolModule {
             revert AddressError.ZeroAddress();
         }
 
-        if (Pool.exists(requestedPoolId)) {
-            revert PoolAlreadyExists(requestedPoolId);
-        }
-
```
Comment (ajsantander) [inline packages/synthetix-main/contracts/modules/core/PoolModule.sol:32]: Moved this requirement into Pool.create()

File: packages/synthetix-main/contracts/modules/core/PoolModule.sol [hunk 2]
```diff
@@ -114,18 +109,18 @@ contract PoolModule is IPoolModule {
 
             for (
                 ;
-                i < (markets.length < pool.poolDistribution.length ? markets.length : pool.poolDistribution.length);
```
Comment (ajsantander) [inline packages/synthetix-main/contracts/modules/core/PoolModule.sol:117]: Renamed Pool's `poolDistribution` (the array of markets and their weight) to `marketConfigurations`

File: packages/synthetix-main/contracts/modules/core/PoolModule.sol [hunk 3]
```diff
@@ -136,26 +131,26 @@ contract PoolModule is IPoolModule {
...
         for (i = 0; i < popped; i++) {
-            Market.rebalance(pool.poolDistribution[pool.poolDistribution.length - 1].market, poolId, 0, 0);
-            pool.poolDistribution.pop();
+            Market.rebalance(pool.marketConfigurations[pool.marketConfigurations.length - 1].market, poolId, 0, 0);
+            pool.marketConfigurations.pop();
         }
 
         pool.totalWeights = uint128(totalWeight);
 
-        pool.rebalanceConfigurations();
```
Comment (ajsantander) [inline packages/synthetix-main/contracts/modules/core/PoolModule.sol:158]: I found that `rebalanceConfigurations` only calls `distributeDebt`. Its an unnecessary wrapper, so I removed it.

File: packages/synthetix-main/contracts/storage/Pool.sol [hunk 4]
```diff
@@ -2,130 +2,227 @@
...
-                (self.totalRemainingLiquidity * weight) / totalWeights
-            );
+            // Note: the factor weight / totalWeights is not deduped below to maintain numeric precision.
+
+            // TODO: Possible CRITICAL BUG - If the int totalAllocatableLiquidity is negative, the casting to uint
+            // below will cause an overflow; a tiny negative number becomes a humongous positive number.
+            // We might need to do a general review of how ints are casted into uints for calculations that involve both types.
+            uint proRataAllocatableLiquidity = (uint(totalAllocatableLiquidity) * weight) / totalWeights;
+            // TODO: Why this? Shouldn't this be an equal and negative values be truncated at totalAllocatableLiquidity?
+            proRataAllocatableLiquidity = totalAllocatableLiquidity > 0 ? proRataAllocatableLiquidity : 0;
```
Comment (ajsantander) [inline packages/synthetix-main/contracts/storage/Pool.sol:149]: Important

Comment (ajsantander) [reply packages/synthetix-main/contracts/storage/Pool.sol:149]: Intriduce SafeCast

File: packages/synthetix-main/contracts/modules/core/PoolModule.sol [hunk 5]
```diff
@@ -29,10 +28,6 @@ contract PoolModule is IPoolModule {
             revert AddressError.ZeroAddress();
         }
 
-        if (Pool.exists(requestedPoolId)) {
-            revert PoolAlreadyExists(requestedPoolId);
```
Comment (noahlitvin) [inline packages/synthetix-main/contracts/modules/core/PoolModule.sol:33]: Why is this removed here?

**Review by noahlitvin — APPROVED** [review]

File: packages/synthetix-main/contracts/modules/core/PoolModule.sol (hunk 5 above)
Comment (ajsantander) [reply packages/synthetix-main/contracts/modules/core/PoolModule.sol:33]: The check was moved from PoolModule to Pool, to here:
https://github.com/Synthetixio/synthetix-v3/pull/1153/files#diff-f5317ceb232ae692021922bfaf85dbbce8bf1208cab93465eaa7a5634e2f8334R118-R121
